controllers/chat/getChannelDetail.js

```diff
@@ -49,7 +65,21 @@ const getChannelDetail = async (req, res) => {
 
 module.exports = getChannelDetail;
 
-const __queryChannelHelper = async (client, channelId, channelType, userId) => {
+const __queryChannelHelper = async (
+  client,
+  channelId,
+  channelType,
+  userId,
+  last_fetch_date = null
+) => {
+  let date_filter = {
+    $lte: new Date()
+  };
+  if (last_fetch_date) {
+    date_filter = {
+      $gte: last_fetch_date
+    };
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The default value for `date_filter` is set to `$lte: new Date()`, but it is immediately overwritten if `last_fetch_date` is provided. This means that the default value is never used. To improve readability and performance, you could remove the default value and only assign `date_filter` once.
```diff
-  let date_filter = {
-    $lte: new Date()
-  };
-  if (last_fetch_date) {
-    date_filter = {
-      $gte: last_fetch_date
-    };
-  }
+  let date_filter = last_fetch_date ? { $gte: last_fetch_date } : { $lte: new Date() };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->